packages/swapper/src/swappers/ZrxSwapper/getZrxTradeQuote/getZrxTradeQuote.ts

```diff
@@ -225,7 +226,80 @@ async function _getZrxPermit2TradeQuote(
     )
   }
 
-  const maybeZrxPriceResponse = await fetchFromZrxPermit2({
+  // If we don't have a wallet, no dice for the permit2 EIP712 data here - but we don't care just yet since we're getting
+  // a *rate* quote, not a *quote* quote on quote quote.
+  if (!hasWallet) {
+    const maybeZrxPriceResponse = await fetchFromZrxPermit2({
+      priceOrQuote: 'price',
```

Comment:
👀

Reply:
it's happening!

Reply:
quote quote quote

![image](https://github.com/user-attachments/assets/9890a011-036a-4c68-8f5a-4d6cfc4b1e5a)